native/src/main/java/io/ballerina/wso2/apim/catalog/ServiceCatalog.java

```diff
@@ -0,0 +1,263 @@
+import static io.ballerina.wso2.apim.catalog.utils.Constants.HTTP_PACKAGE_NAME;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.createMd5Hash;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.generateBasePath;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getDefinitionType;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getHostname;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getHttpAnnotation;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getModuleAnnotation;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getOpenApiDefinition;
+import static io.ballerina.wso2.apim.catalog.utils.Utils.getSecurityType;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.COLON;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.CONFIG;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFAULT_STRING;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_FILE_CONTENT;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_TYPE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DEFINITION_URL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.DESCRIPTION;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.LOCALHOST;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MD5;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MUTUAL_SSL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.MUTUAL_SSL_ENABLED;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.NAME;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.NONE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.PORT;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SECURE_SOCKET;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SECURITY_TYPE;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SERVICE_KEY;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.SERVICE_URL;
+import static io.ballerina.wso2.apim.catalog.utils.Constants.VERSION;
+
+/**
+ * Provide functionality to retrieve artifacts details.
+ *
+ * @since 0.1.0
+ */
+public class ServiceCatalog {
+
+    public static BArray getArtifacts(Environment env) {
+        Module currentModule = env.getCurrentModule();
+        RecordType recordType = TypeCreator.createRecordType(Constants.SERVICE_ARTIFACT_TYPE_NAME,
+                currentModule, 0, false, 0);
+        ArrayType arrayType = TypeCreator.createArrayType(recordType);
+        BArray arrayValue = ValueCreator.createArrayValue(arrayType);
+
+        for (Artifact artifact : env.getRepository().getArtifacts()) {
+            Object serviceObj = artifact.getDetail(Constants.SERVICE);
+            Type originalType = ((BObject) serviceObj).getOriginalType();
+            Module module = originalType.getPackage();
+            if (module != null && module.equals(currentModule)) {
```

Comment:
Assuming current module is not null I think using `Objects.equals` is cleaner

```suggestion
            if (Objects.equals(module, currentModule)) {
```

Reply:
I think current way is more clear about the intention